### PR TITLE
add rcon-rest-api container

### PR DESCRIPTION
**Pull request by Kariton**
[nekomeowww/factorio-rcon-api](https://github.com/nekomeowww/factorio-rcon-api)
> Fully implemented wrapper for Factorio headless server console as RESTful and gRPC for easier management through APIs

This is a fairly new project i discovered by chance.
hope you dont mind adding a additional (optional) container which adds so much value (or has the potential to do so).

some quick examples:
```sh
# curl -X GET https://factorio/api/v1/factorio/console/command/version
{"version":"2.0.11"}
# curl -X GET https://factorio/api/v1/factorio/console/command/seed
{"seed":"3508092348"}
# curl -X GET https://factorio/api/v1/factorio/console/command/players
{"players":[{"username":"Players","online":false},{"username":"MYNAME","online":true}]} # '{"username":"Players","online":false}' look bugged to me. i am alone; or am i?
# curl -X GET https://factorio/api/v1/factorio/console/command/evolution
{"evolutionFactor":0.0202,"time":100,"pollution":0,"spawnerKills":0}
# curl -X POST https://factorio/api/v1/factorio/console/command/server-save
{}
# curl -X POST https://factorio/api/v1/factorio/console/command/kick -d '{"username": "nicname","reason": "API Test"}'
{}
```

### CODE REVIEW

**Comment by Kariton** [comment]
gonna add new livenessprobes later. `/healthz` is supported now.

**Comment by Kariton** [comment]
same as in my other pr...
> Error: Service "factorio-server-charts-qqm09zfbha-rcon" is invalid: spec.ports[0].targetPort: Invalid value: "27015": must contain at least one letter (a-z)

i dont even know what this is talking about. ^^
are we now an harbor? where ports have letters?

**Comment by Kariton** [comment]
oi. `quote`

**Comment by Kariton** [comment]
as known:
> cli: execute: auth: rcon: dial tcp 10.96.122.250:30100: connect: connection refused

**Comment by SQLJames** [comment]
@Kariton there are some conflicts due to the port fixer merge, do you want to update this and I'll begin QA on this as well?

**Comment by Kariton** [comment]
hm. github merge resolver merges main onto PR branch.
should i rebase (and squash) instead?

**Comment by Kariton** [comment]
i thing disabling tls per default is better because it requires further configuration anyways.

File: charts/factorio-server-charts/values.yaml [hunk 1]
```diff
@@ -339,6 +344,42 @@ rcon:
...
+## @param rcon_api.serviceAnnotations Annotations to add to the RCON API service
+## @param rcon_api.httpPort Internal port on which the HTTP API for RCON is exposed
+## @param rcon_api.grpcPort Internal port on which the gRPC API for RCON is exposed
+## @param rcon_api.ingress.enabled Enable or disable the Ingress for the RCON API service
+## @param rcon_api.ingress.hostname Hostname for the RCON API Ingress
+## @param rcon_api.ingress.className Ingress class name for selecting the Ingress controller (e.g., nginx, cilium)
+## @param rcon_api.ingress.annotations Additional annotations for the Ingress resource
+## @param rcon_api.ingress.tls.enabled Enable or disable TLS for the Ingress
+## @param rcon_api.ingress.tls.secretName Kubernetes Secret name for TLS certificates used by the Ingress
+rcon_api:
```
Comment (SQLJames) [inline charts/factorio-server-charts/values.yaml:361]: Can we update this to be rconAPI?

**Review by SQLJames — COMMENTED** [review]
Left a comment about making the values rconAPI

File: charts/factorio-server-charts/Chart.yaml [hunk 2]
```diff
@@ -20,7 +20,7 @@ sources:
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 2.1.0
+version: 2.1.1
```
Comment (SQLJames) [inline charts/factorio-server-charts/Chart.yaml:23]: This should be 2.2.0 since it's new functionality added to the chart in a backwards compatible way.

**Comment by SQLJames** [comment]
Also left comment for chart version

**Review by SQLJames — APPROVED** [review]

**Pull request merged**